app/screens/root/root.js

```diff
@@ -1,28 +1,33 @@
 // Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
 // See License.txt for license information.
 
-import React, {PureComponent} from 'react';
+import React, {Component} from 'react';
 import PropTypes from 'prop-types';
 
 import {Client, Client4} from 'mattermost-redux/client';
-import {RequestStatus} from 'mattermost-redux/constants';
 
 import Loading from 'app/components/loading';
 import {stripTrailingSlashes} from 'app/utils/url';
 
-export default class Root extends PureComponent {
+export default class Root extends Component {
     static propTypes = {
         allowOtherServers: PropTypes.bool,
         credentials: PropTypes.object,
         justInit: PropTypes.bool,
-        loginRequest: PropTypes.object,
         navigator: PropTypes.object,
         theme: PropTypes.object,
         actions: PropTypes.shape({
             loadMe: PropTypes.func
         }).isRequired
     };
 
+    shouldComponentUpdate(nextProps) {
```

Comment:
Why would anything change here other than the credentials?

Reply:
this is what happens if we change it to a pure component

![image](https://user-images.githubusercontent.com/6757047/31451508-b04d0d58-ae82-11e7-92a1-55abfad07a2e.png)
![image](https://user-images.githubusercontent.com/6757047/31451524-b8889d70-ae82-11e7-9f86-6ed4c91de411.png)

Reply:
I'm not sure what that's saying. Is it saying the update is triggering because the navigator changed? Or the theme changed?

Reply:
What's going on here is that somehow the shallow check is failing and the only thing we really care about here changing is the credentials, so...